src/gathernodeinputhandler.cpp

```diff
@@ -0,0 +1,74 @@
+//*****************************************************************************
+// Copyright 2021 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "gathernodeinputhandler.hpp"
+
+#include "logging.hpp"
+#include "ov_utils.hpp"
+
+namespace ovms {
+
+GatherNodeInputHandler::GatherNodeInputHandler(uint32_t inputsMissingCount, session_id_t shardsCount) :
+    NodeInputHandler(inputsMissingCount) {
+    expectedDependencies *= shardsCount;
+}
+
+void GatherNodeInputHandler::setInput(const std::string& inputName, InferenceEngine::Blob::Ptr& ptr, session_id_t shardId) {
+    auto it = shardsStorage.find(inputName);
```

Comment:
Here you access the same storage with 2 different types: inputName(std::string) and shardId in line 31 shardId(session_id_t) - this might lead to confusion, from my understanding session_id_t is std::string?